Wangscape/TilePartitionerBase.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <vector>
+#include <SFML/Graphics.hpp>
+#include <functional>
+#include "Options.h"
+
+class TilePartitionerBase
+{
+public:
+    typedef std::vector<std::pair<sf::Texture, TerrainID>> TilePartition;
+    typedef std::vector<TerrainID> Corners;
```

Comment:
Oh, another `typedef`. There are too many of them for me. It seems for that it will be better to reduce the numbers of `typedef`s and move them to separate files.

I like to treat them like classes (new types) and put one `typedef` in one file (not always, of course). It could seem to you that this way there will too many files with two or three lines, but in my opinion it's just matter of dividing source code into modules. In Wangscape we have roughly (as I noticed):
* Reader
  - JSON options
  - command line arguments
  - input images
* Generator
  - current tileset generator
  - modifiers
    + borders
    + **Perlin noise**
* Writer
  - output images
  - output JSONs

We could (and I think we should) map it to a directory structure and namespaces.